api.go

```diff
@@ -111,6 +112,7 @@ func NewApiWithClient(a ma.Multiaddr, c *gohttp.Client) (*HttpApi, error) {
 	api := &HttpApi{
```

Comment:
We could just call `NewURLApiWithClient` here.

Reply:
Good idea!